2600hz/kazoo
# [master] fix duplication in search view, fix search_by_name missing variable error

Fixing duplication result issue in `search_by_name_and_number` view. Previously it set name to `pvt_alphanum_name` and emit, and also emits another one if the `doc.name` is defined.

Generalized name, first_name and last_name logic in both "search by name and number" and search by name view to do:

1. If either first or last name is defined, sanitize and concatenate them, and store non sanitized one to set in the value part of the result (so UI can show this).
2. If doc has `pvt_alphanum_name` key use that for name part of the result key.
    a. else if the `name` key defined sanitize it and use it for key, and use non sanitized for value
    b. if all failed and `name` (from concatenation of the first and last name is empty) is empty use the ID of document.
3. set the value object of whatever found for name, first/last name (if any), or numbers (if any, in name_and_number view).
4. search by name and number is emitting one result for name and for each numbers (if any).

4.2: https://github.com/2600hz/kazoo/pull/4945


### How To Test

Using UI, use search box in account manager, account switch at top bar, use search in advanced callflow. Check the search result using Web Developer tools, the result should not be duplicated.
